### ISSUE TITLE
Feature Request:TOE Layered Icon Changes

### ISSUE THREAD

**Issue by trent1098s** [MegaMek/mekhq#347]
Three simple requests in one;

#1 In the XML file for the campaign the items selected on each layer are stored in the XML file. Can a change be made to the Change Force Icon context menu code so that when the dialog opens, the _currently selected layers_ are pre-populated? Currently if you build a layered icon then edit the layered icon you have to start from scratch again. (This is aggravating if you have a 4 or 5 layer icon and just needed to tweak the Formation, or whatever, you have to go from tab to tab rebuilding the icon from scratch)

#2 A "none" selection in each tab for building the layered icon. Currently if you accidentally click/tap a section (such as select a background image) there is no way of clearing that out other than starting all over again.

#3 Add "Copy Icon" to the context menu and "Paste Icon" to the context menu on the TOE page. This would allow you to more quickly build a TOE by copying an icon from one force to another. 

These three things would rapidly accelerate the TOE building process for larger forces. I'm running a campaign where there are four regimental sized units on the planet plus air wing and associated other units, and it took 8 hours just to build the icons for the TOE on the units. :)

**Comment by Dylan-M** [comment]
1. This is a known limitation to an unfinished feature. It isn't trivial to do.

2. You can clear it using ctrl-click for most, if not all, selections. A "None" is planned.

3. This isn't trivial either.

Regardless, these should be three separate tickets. Please see:
https://github.com/MegaMek/megamek/wiki/Making-a-Bug-Report

Specifically the section regarding One Issue Per Ticket

**Comment by trent1098s** [comment]
Thanks, in the future I'll split up requests to multiple tickets. 

It doesn't sound like there's interest in doing these changes; I can probably just write something in Visual Studio that lets me manipulate these campaign files outside of mekHQ, so I can set up the icons in bulk. It's a real time consuming nightmare trying to do a 'prettied up' regiment size unit in the current interface.

For completeness this is the data relating to the above in the campaign file.

						<force id="81" type="mekhq.campaign.force.Force">
							<name>1st Squadron</name>
							<desc></desc>
							<iconCategory>Layered</iconCategory>
							<iconHashMap>
								<iconentry key="Pieces/Frames/">
									<value name="Frame.png"/>
								</iconentry>
								<iconentry key="Pieces/Type/">
									<value name="Vehicle Tracked.png"/>
								</iconentry>
								<iconentry key="Pieces/Formations/">
									<value name="08 Battalion.png"/>
								</iconentry>
								<iconentry key="Pieces/Backgrounds/">
									<value name="Free Worlds League.png"/>
								</iconentry>
							</iconHashMap>

**Comment by Dylan-M** [comment]
It isn't lack of interest, simply lack of time for me currently. These are issues that should be addressed. Well, with the exception of 3, which is a nice to have. It can definitely be done, it's just lack of time for me right now. It doesn't help that I'm trying to get the last piece of our website transferred to the new server and having internet troubles while doing it :)

**Comment by trent1098s** [comment]
Understood. I've been doing development work for the last few decades so I know how triage on feature requests work. If I wasn't buried under work in the day job I'd jump in and lend a hand but by the time I'm off work the last thing I want to think about is writing code. A decade or two ago I'd be all over it, but I got jaded as I got older. :)

As it stands it should be pretty trivial for me to write a quickie in C# that'll spit out the required XML that I can just drop in to the campaign file (or if I took a little more time could just have the program edit the xml file directly). No problem with this staying low priority while you tackle bigger stuff. :)

(In all honesty there's way bigger things I'd like to have too!)
